actor/supervisor.go

```diff
@@ -0,0 +1,98 @@
+package actor
+
+import (
+	"fmt"
+	"sync"
+)
+
+type SupervisedWorker struct {
+	Worker
+	Inbox  <-chan Message
+	Outbox chan<- Message
+}
+
+func (sw *SupervisedWorker) SetInbox(inbox <-chan Message) {
+	sw.Inbox = inbox
+}
+
+func (sw *SupervisedWorker) SetOutbox(outbox chan<- Message) {
+	sw.Outbox = outbox
+}
+
+type supervisedWorker interface {
+	Worker
+	GetID() string
+	SetInbox(inbox <-chan Message)
+	SetOutbox(outbox chan<- Message)
+}
+
+type Supervisor struct {
```

Comment:
Could `Supervisor` be implemented using `Actor` and `Worker` logic?

- `Supervisor` currently has `StartAll` and `StopAll` methods, those could be easily changed to comply with `Actor` interface. 
- Could work logic of supervisor be implemented with `Worker` interface?